mechanisms/CMakeLists.txt

```diff
@@ -31,11 +31,22 @@ make_catalogue(
   STANDALONE FALSE
   VERBOSE ${ARB_CAT_VERBOSE})
 
+make_catalogue(
+  NAME empty
+  SOURCES "${CMAKE_CURRENT_SOURCE_DIR}/empty"
+  OUTPUT "CAT_EMPTY_SOURCES"
+  MECHS
+  PREFIX "${PROJECT_SOURCE_DIR}/mechanisms"
+  CXX_FLAGS_TARGET "${ARB_CXX_FLAGS_TARGET_FULL}"
+  STANDALONE FALSE
+  VERBOSE ${ARB_CAT_VERBOSE})
+
 # Join sources
 set(arbor_mechanism_sources
   ${CAT_BBP_SOURCES}
   ${CAT_ALLEN_SOURCES}
   ${CAT_DEFAULT_SOURCES}
+  ${CAT_EMPTY_SOURCES}
```

Comment:
But you are making this way harder than it should be. Just implement
```cxx
const mechanism_catalogue& global_empty_catalogue() {
  static mechanism_catalogue empty = mechanism_catalogue{};
  return empty;
}
```

Reply:
Where should this code go?

Reply:
This should go into `mechcat.cpp`, like the rest of the `global_XYZ_catalogue` functions. 

As for the extensions:
- this returns a constant reference thus immutable
- if you want to extend it, you'll have to make a mutable copy

Reply:
I don't understand the part of the extensions, will this work:

```python
e1 = arbor.empty_catalogue()
e2 = arbor.empty_catalogue()
e1.extend(smth)
assert len(e2) == 0
```

Reply:
Like this?
```py
In [1]: import arbor as A

In [2]: c = A.default_catalogue()

In [4]: len([n for n in c])
Out[4]: 9

In [5]: c.extend(A.default_catalogue(), 'cc/')

In [6]: len([n for n in c])
Out[6]: 18
```

Reply:
After reading through the sources again, I think the easiest course to take is to expose the default constructor of `mechanism_catalogue` to python. See here
https://github.com/arbor-sim/arbor/blob/f13f786a35f07c6eb00f7d6d18c07a2c16664f7a/python/mechanism.cpp#L155
only the copy constructor is bound, simply add the default one in the same fashion.

As for tests, go to hereabouts 
https://github.com/arbor-sim/arbor/blob/f13f786a35f07c6eb00f7d6d18c07a2c16664f7a/python/test/unit/test_catalogues.py#L73
and add tests for the behaviours you rely on, here's the list I would currently see 
1. `A.catalogue()` has no mechs
2. `A.catalogue().extend(other)` has `other`'s mechs
3. and so on

The docs here
https://github.com/arbor-sim/arbor/blob/master/doc/python/mechanisms.rst#mechanism-catalogues
need a line or two on the constructors.

Reply:
I dont' understand this in the slightest 😇 What I can make of Line 155 in f13f786 is that the constructor is already exposed? And I don't see anything mentioning `copy`. Perhaps it's better you implement this.

Reply:
Welcome to C++ :) A constructor with a signature like `A(const A&)` is called a copy ctor (for `A`).
We have also the additional pleasure of working with pybind11 here, so 
```cxx
pybind11::class_<arb::mechanism_catalogue> cat(m, "catalogue");
cat.def(pybind11::init<const arb::mechanism_catalogue&>())
```
means: on the class `cat`, define a constructor (by way of `init`) that takes a `const arb::mechanism_catalogue&`
for arguments. Effectively a copy ctor.
```cxx
cat.def(pybind11::init())
```
should do the trick.